var/spack/repos/builtin/packages/py-rpy2/package.py

```diff
@@ -10,39 +10,43 @@ class PyRpy2(PythonPackage):
     interface to R from Python, a proposed high-level interface,
     including wrappers to graphical libraries, as well as R-like
     structures and functions.
-
     """
 
+    homepage = "https://rpy2.github.io"
     pypi = "rpy2/rpy2-2.5.4.tar.gz"
 
     license("GPL-2.0-or-later")
 
+    maintainers("Chrismarsh")
+
+    version("3.5.17", sha256="dbff08c30f3d79161922623858a5b3b68a3fba8ee1747d6af41bc4ba68f3d582")
     version("3.0.4", sha256="2af5158a5d56af7f7bf5e54d8d7e87b6f115ff40f056d82f93cad0cbf6acc0cb")
     version("3.0.0", sha256="34efc2935d9015527837d6b1de29641863d184b19d39ad415d5384be8a015bce")
-    version("2.9.4", sha256="be57f741d0c284b5d8785ab03dff0e829303e5ac30e548d5ceb46e05b168812e")
-    version("2.8.6", sha256="004d13734a7b9a85cbc1e7a93ec87df741e28db1273ab5b0d9efaac04a9c5f98")
-    version("2.5.6", sha256="d0d584c435b5ed376925a95a4525dbe87de7fa9260117e9f208029e0c919ad06")
-    version("2.5.4", sha256="d521ecdd05cd0c31ab017cb63e9f63c29b524e46ec9063a920f640b5875f8a90")
 
-    # FIXME: Missing dependencies:
-    # ld: cannot find -licuuc
-    # ld: cannot find -licui18
+    variant("numpy", default=True, description="Numpy", when="@3.5.17:")
+    variant("pandas", default=True, description="Pandas", when="@3.5.17:")
 
-    # All versions
     depends_on("py-setuptools", type="build")
-    depends_on("r", type=("build", "run"))
+
+    with when("@3.5.17:"):
```

Comment:
Did these dependencies start at 3.5.17, or at a previous minor version e.g. 3.5? With python packages, the with when blocks often lead to hard to maintain structures. Easier to have the blocks as used by the template:
- python lines (and when clauses where new python versions become requires)
- build system lines
- active dependencies
- historical dependecies

Reply:
I tried taking a look at this, but quickly ended up in untested depedency hell as patch versions would bump versions. I've deprecated the 6 year old 3.0.x versions, added support for python 3.9,  I've removed the when, and made the 3.5.17 flat to support future version additions.